### PR TITLE
[BD-46] feat: update React version, fixed tests

### PR DESCRIPTION
**Pull request by PKulkoRaccoonGang**
## Description

Update React version, fixed tests.

JIRA: [PAR-772](https://openedx.atlassian.net/browse/PAR-772)

## Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [x] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [ ] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [x] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [x] Is there adequate test coverage for your changes?
* [ ] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [x] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [x] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [x] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @PKulkoRaccoonGang! I've created [BLENDED-1211](https://openedx.atlassian.net/browse/BLENDED-1211) to keep track of it in Jira.   More details are on the [BD-46](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3134685276/BD-46+Paragon+Blended+Support+and+Maintenance) project page.


When this pull request is ready, tag your edX technical lead.

File: package.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
     "font-awesome": "^4.7.0",
     "lodash.uniqby": "^4.7.0",
     "mailto-link": "^1.0.0",
-    "prop-types": "^15.8.1",
+    "prop-types": "^15.7.2",
```
Comment (viktorrusakov) [inline package.json:48]: Why do we need to downgrade `prop-types` version?

File: package.json [hunk 2]
```diff
@@ -89,9 +88,10 @@
     "jest": "^25.0.0",
     "jest-cli": "^25.0.0",
     "markdown-loader-jest": "^0.1.1",
-    "react": "^16.14.0",
-    "react-dom": "^16.14.0",
-    "react-test-renderer": "^16.14.0",
+    "prop-types": "^15.8.1",
```
Comment (viktorrusakov) [inline package.json:91]: We already have `prop-types` package in dependencies, I don't think we need it as a dev dependency too

**Comment by viktorrusakov** [comment]
Also I feel like there is something wrong with your `package-lock` file 🤔 There seems to be too many additions, I'm not sure if upgrading to React 17 would cause such changes

**Comment by adamstankiewicz** [comment]
> Also I feel like there is something wrong with your `package-lock` file 🤔 There seems to be too many additions, I'm not sure if upgrading to React 17 would cause such changes.

@viktorrusakov I think this is because the current package-lock.json file seems to be using `lockfileVersion: 1` even though Paragon is supposedly running on Node 16 now. I think we're seeing so many additions because the install was done using Node 16, forcing our `lockfileVersion: 1` to `lockfileVersion: 2` as it should be.

Are you running Node 16 locally, @viktorrusakov?

File: src/SelectableBox/index.jsx [hunk 3]
```diff
@@ -47,16 +47,17 @@ const SelectableBox = React.forwardRef(({
...
 
-  const onClickHandler = () => {
-    inputRef.current.click();
-    if (onClick) { onClick(inputRef.current); }
-  };
+  useEffect(() => {
+    if (onClick && inputRef.current) {
+      inputRef.current.onclick = () => onClick(inputRef.current);
+    }
+  }, [inputRef.current]);
```
Comment (adamstankiewicz) [inline src/SelectableBox/index.jsx:54]: [clarification] Can you tell me a bit more about why this change was necessary?

File: package.json (hunk 1 above)
Comment (PKulkoRaccoonGang) [reply package.json:48]: We don't need to downgrade prop-types, fixed

File: package.json (hunk 2 above)
Comment (PKulkoRaccoonGang) [reply package.json:91]: I agree, remove

File: src/SelectableBox/index.jsx (hunk 3 above)
Comment (PKulkoRaccoonGang) [reply src/SelectableBox/index.jsx:54]: After upgrading the version of the adapter, there was an inaccuracy in the test coverage of the SelectableBox component. enzyme-adapter-react-17 did track double clicks, while its older version, enzyme-adapter-react-16, did not. Refactoring the SelectableBox component helped us fix the bug we found and make sure the component works correctly.

**Review by adamstankiewicz — APPROVED** [review]
Looks good to me! Will unfortunately need a rebase, though.

**Comment by openedx-webhooks** [comment]
@PKulkoRaccoonGang 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/1248&entry.1484655318=https://github.com/PKulkoRaccoonGang&entry.752974735=2022-04-21+20:13&entry.1917517419=2022-05-06+09:51&entry.2133058324=Yes) so we can improve your experience in the future.
